source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,293 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+On this page you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an irreducible series of one or
+more database operations such that all occur or none occur.
```

Comment:
Thanks for the note. I believe the proposed definition involving "interruptions" is slightly incorrect, as I believe atomic operations still undergo context switching on the OS (and thus are periodically interrupted). I don't know how to defend this claim with primary evidence without digging into the MongoDB server codebase, but in the implementation section of the Wikipedia article on [Database Atomicity](https://en.wikipedia.org/wiki/Atomicity_(database_systems)#Implementation) it states that "Portable operating systems cannot simply block interrupts to implement synchronization, since hardware that lacks concurrent execution such as hyper-threading or multi-processing is now extremely rare." In this quote, I am assuming that portable is used as in Portable Operating System Interface (POSIX).

Because of the above argument, I believe the following definition provided by Jeff Allen is more accurate (there is now a ticket on server for adding this definition to the glossary).

> An atomic operation is an operation which either completes entirely, or does not complete at all. Atomic operations cannot partially complete.

Update to new definition: